app/Console/Commands/Takedate.php

```diff
@@ -29,80 +38,140 @@ class Takedate extends Command
-			$i++;
+			$this->progressBar->advance();
+			// TODO: As soon as we support AWS S3 storage, we must stop using absolute paths. However, first the EXIF extractor must be rewritten to use file streams.
+			$fullPath = $photo->size_variants->getOriginal()->getFile()->getAbsolutePath();
+
 			if (!file_exists($fullPath)) {
-				$this->line($i . ': File ' . $fullPath . ' not found for ' . $photo->title . '.');
+				$this->printError('File ' . $fullPath . ' not found for photo "' . $photo->title . '" (ID=' . $photo->id . ').');
 				continue;
 			}
-			$info = $metadataExtractor->extract($fullPath, $photo->type);
-			/* @var \DateTime $stamp */
+
+			$kind = $photo->isRaw() ? 'raw' : ($photo->isVideo() ? 'video' : 'photo');
+			$info = $metadataExtractor->extract($fullPath, $kind);
+			/* @var Carbon $stamp */
 			$stamp = $info['taken_at'];
-			if ($stamp != null) {
-				if ($stamp == $photo->takestamp) {
-					$this->line($i . ': Takestamp up to date for ' . $photo->title);
+			if ($stamp !== null) {
+				// Note: `equalTo` only checks if two times indicate the same
+				// instant of time on the universe's timeline, i.e. equality
+				// comparison is always done in UTC.
+				// For example "2022-01-31 20:50 CET" is deemed equal to
+				// "2022-01-31 19:50 GMT".
+				// So, we must check for equality of timezones separately.
+				if ($photo->taken_at->equalTo($stamp) && $photo->taken_at->timezoneName === $stamp->timezoneName) {
+					$this->printInfo('Takestamp ' . $stamp->format(self::DATETIME_FORMAT) . ' up to date for photo "' . $photo->title . '" (ID=' . $photo->id . ').');
 					continue;
 				}
 				$photo->taken_at = $stamp;
 				if ($photo->save()) {
-					$this->line($i . ': Takestamp updated to ' . $stamp->format('d M Y \a\t H:i') . ' for ' . $photo->title);
+					$this->printInfo('Takestamp updated to ' . $stamp->format(self::DATETIME_FORMAT) . ' for photo "' . $photo->title . '" (ID=' . $photo->id . ').');
 				} else {
-					$this->line($i . ': Failed to update takestamp for ' . $photo->title);
+					$this->printError('Failed to update takestamp for photo "' . $photo->title . '" (ID=' . $photo->id . ').');
 				}
 				continue;
 			}
-			if (!$timestamps) {
-				$this->line($i . ': Failed to get Takestamp data for ' . $photo->title . '.');
+
+			// TODO: The description of this command's signature suggests something different than we do here.
+			// The description suggests that we would use the file date as a fall back for `taken_at`, if EXIF data is not available.
+			// However, we set `created_at` (aka upload time) here.
+			// The latter is actually an independent operation of the first.
+			if (!$useFileTime) {
+				$this->printError('Failed to get takestamp data for photo "' . $photo->title . '" (ID=' . $photo->id . ').');
 				continue;
 			}
 			if (is_link($fullPath)) {
 				$fullPath = readlink($fullPath);
 			}
 			$created_at = filemtime($fullPath);
 			if ($created_at == $photo->created_at->timestamp) {
```

Comment:
We don't need to compare the timezones here like you do for `taken_at`? earlier

Reply:
`created_at` (and `updated_at`) only store the datetime in UTC without any timezone indication. So it is sufficient to compare the timestamp. This is also the only information we get from the filesystem anyway.

We only additionally store the original timezone in `taken_at_orig_tz` alongside `taken_at` in UTC so that we can present the user a datetime which matches the user's expectation when they look at the photo (otherwise a photo taken during broad daylight in Australia would show a midnight timestamp).